packages/datadog-plugin-grpc/src/client.js

```diff
@@ -0,0 +1,181 @@
+'use strict'
+
+const Tags = require('../../../ext/tags')
+const TEXT_MAP = require('../../../ext/formats').TEXT_MAP
+const kinds = require('./kinds')
+const { addMethodTags, addMetadataTags, getFilter } = require('./util')
+
+function createWrapMakeClientConstructor (tracer, config, grpc) {
+  config = config.client || config
+
+  return function wrapMakeClientConstructor (makeClientConstructor) {
+    return function makeClientConstructorWithTrace (methods) {
+      const ServiceClient = makeClientConstructor.apply(this, arguments)
+      const proto = ServiceClient.prototype
+
+      Object.keys(methods)
+        .forEach(method => {
+          const originalName = methods[method].originalName
+
+          proto[method] = wrapMethod(tracer, config, proto[method], methods[method], grpc)
+
+          if (originalName) {
+            proto[originalName] = wrapMethod(tracer, config, proto[originalName], methods[method], grpc)
```

Comment:
Is there a way to refactor this so we can just assign the same `methodWithTrace` that we assigned to `proto[method]` to `proto[originalName]`. The reason is `gRPC` [does something similar](https://github.com/grpc/grpc-node/blob/2ab15fef4ff19d02b78f33a0290ca10f6978262b/packages/grpc-js/src/make-client.ts#L140).

Reply:
I also tested this with a weird case: if you have both a camelCased and snake_cased method defined, the app just doesn't work properly, so the previous should be okay to do I think.